tgui/packages/tgui/interfaces/LocalGenerator/index.tsx

```diff
@@ -0,0 +1,186 @@
+  };
+
+  const handleEjectTank = () => {
+    act("eject-tank");
+  };
+
+  const handleEjectCell = () => {
+    act("eject-cell");
+  };
+
+  const handleConnectAPC = () => {
+    act("connect-APC");
+  };
+
+  return (
+    <Window title={name} width={300} height={470}>
+      <Window.Content>
+        <Stack vertical>
+          <Section title="Generator">
+            <Stack vertical>
+              <Stack.Item>
+                <Stack align="baseline">
+                  <Stack.Item basis="50%">
+                    <Button
+                      width={11}
+                      content="Toggle Floor Bolts"
+                      onClick={() => handleToggleBolts()} />
+                  </Stack.Item>
+                  <Stack.Item>
+                    Status:
+                  </Stack.Item>
+                  {boltsStatus ? (
+                    <Stack.Item color="good">
+                      ACTIVE
+                    </Stack.Item>
+                  ) : (
+                    <Stack.Item color="average">
+                      INACTIVE
+                    </Stack.Item>
+                  )}
+                </Stack>
+              </Stack.Item>
+              <Stack.Item>
+                <Stack align="baseline">
+                  <Stack.Item basis="50%">
+                    <Button
+                      width={11}
+                      content="Toggle Generator"
+                      onClick={() => handleToggleGenerator()} />
+                  </Stack.Item>
+                  <Stack.Item>
+                    Status:
+                  </Stack.Item>
+                  {generatorStatus ? (
+                    <Stack.Item color="good">
+                      ACTIVE
+                    </Stack.Item>
+                  ) : (
+                    <Stack.Item color="average">
+                      INACTIVE
+                    </Stack.Item>
+                  )}
+                </Stack>
+              </Stack.Item>
+              <Stack.Item>
+                <Stack align="baseline">
+                  <Stack.Item basis="50%">
+                    <Button
+                      width={11}
+                      content="Change Target"
+                      onClick={() => handleSwapChargeTarget()} />
+                  </Stack.Item>
+                  <Stack.Item>
+                    Target:
+                  </Stack.Item>
+                  {chargeAPC ? (
+                    <Stack.Item color="good">
+                      APC
+                    </Stack.Item>
+                  ) : (
+                    <Stack.Item color="good">
+                      CELL
+                    </Stack.Item>
+                  )}
+                </Stack>
+              </Stack.Item>
+            </Stack>
+          </Section>
+          <PortableHoldingTank
+            holding={holding}
+            onEjectTank={handleEjectTank} />
+          <Section
+            title="Internal Cell"
+            height={7.25}
```

Comment:
Shouldn't be needed (i.e. should be able to be derived from content)? Ditto for similar. I suppose there's an argument for having the same height whether the section has content or not to make the layout the same, in which case not overly opposed - it's a small enough UI that it doesn't really matter.